module/database/UserDatabase.py

```diff
@@ -63,23 +75,24 @@ def changePassword(db, user, oldpw, newpw):
 
         salt = r[2][:5]
         pw = r[2][5:]
-        h = sha1(salt + oldpw)
+        h = sha1(salt.encode('utf-8') + oldpw.encode('utf-8'))
         if h.hexdigest() == pw:
-            salt = reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
-            h = sha1(salt + newpw)
+            salt = functools.reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
+            h = sha1(salt.encode('utf-8') + newpw.encode('utf-8'))
```

Comment:
Codacy found an issue: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=4846282)